tanjun/repeaters.py

```diff
@@ -0,0 +1,425 @@
+
+        Examples
+        --------
+        ```py
+        @component.with_repeater(
+            delay=1,
+            max_runs=20
+        )
+        async def repeater():
+            global counter
+            counter += 1
+            print(f"Run #{counter}")
+
+
+        @repeater.with_pre_callback
+        async def pre():
+            print("pre callback")
+        ```
+        """
+        self._pre_callback = callback
+        return callback
+
+    def with_post_callback(self, callback: CallbackSig) -> CallbackSig:
+        """Set the callback executed after the repeater is finished.
+
+        Parameters
+        ----------
+        callback : CallbackSig
+            The callback to set.
+
+        Returns
+        -------
+        CallbackSig
+            The callback.
+
+        Examples
+        --------
+        ```py
+        @component.with_repeater(
+            delay=1,
+            max_runs=20
+        )
+        async def repeater():
+            global counter
+            counter += 1
+            print(f"Run #{counter}")
+
+
+        @repeater.with_post_callback
+        async def post():
+            print("pre callback")
+        ```
+        """
+        self._post_callback = callback
+        return callback
+
+    def set_ignored_exceptions(self, *exceptions: type[Exception]) -> Repeater[CallbackSigT]:
+        """
+        Set the exceptions that a task will ignore.
+
+        If any of these exceptions are encountered, there will be nothing printed to console
+
+        Parameters
+        ----------
+        exceptions : list[type[Exception]]
+            List of exception types
+
+        Returns
+        -------
+        Repeater[CallbackSigT]
+            Self
+        """
+        self._ignored_exceptions = list(exceptions)
+        return self
+
+    def set_fatal_exceptions(self, *exceptions: type[Exception]) -> Repeater[CallbackSigT]:
+        """
+        Set the exceptions that will stop a task.
+
+        If any of these exceptions are encountered, the task will stop.
+
+        Parameters
+        ----------
+        exceptions : list[type[Exception]]
+            List of exception types
+
+        Returns
+        -------
+        Repeater[CallbackSigT]
+            Self
+        """
+        self._fatal_exceptions = list(exceptions)
+        return self
+
+
+def with_ignored_exceptions(
+    *exceptions: type[Exception],
+) -> collections.abc.Callable[[Repeater[CallbackSigT]], Repeater[CallbackSigT]]:
+    """
+    Set the exceptions that a task will ignore.
+
+    If any of these exceptions are encountered, there will be nothing printed to console.
+
+    .. note:: Even if an exception is ignored, it will stop that iteration of the repeater.
+
+    Parameters
+    ----------
+    exceptions : list[type[Exception]]
+        List of exception types
+
+    Examples
+    --------
+    ```py
+    @tanjun.with_ignored_exceptions(ZeroDivisionError)
+    @tanjun.as_repeater(seconds=1)
```

Comment:
these examples are how this should look but should also include the `@component.with_repeater` call